src/io/github/sspanak/tt9/ui/UI.java

```diff
@@ -46,4 +48,13 @@ public static void toastLong(Context context, int resourceId) {
 	public static void toastLong(Context context, CharSequence msg) {
 		Toast.makeText(context, msg, Toast.LENGTH_LONG).show();
 	}
+
+	public static void toastLang(Context context, CharSequence msg) {
+		if (langToast != null) {
+			langToast.cancel();
+		}
+
+		langToast = Toast.makeText(context, msg, Toast.LENGTH_SHORT);
```

Comment:
Is there any need to re-create the Toast every time? Isn't this more efficient:
```suggestion
		if (langToast == null) {
			langToast = Toast.makeText(context, msg, Toast.LENGTH_SHORT);
		} else {
			langToast.cancel();
		}

```
?

Btw, I've been trying to get rid of these toasts, but they keep coming back... :disappointed:

Reply:
I tried a few different solutions here. If we reuse the toast, sometimes they don't appear (about 10% of the time), even when they are explicitly cancelled beforehand.

> Btw, I've been trying to get rid of these toasts, but they keep coming back... disappointed

I gathered that much from the recent commits, but couldn't come up with a better idea :grimacing:

Reply:
Yeah, well... let's keep it. I can't come up with anything better either.